presto-hive/src/main/java/com/facebook/presto/hive/metastore/ThriftHiveMetastore.java

```diff
@@ -582,6 +591,16 @@ public void grantTablePrivileges(String databaseName, String tableName, String g
     public void revokeTablePrivileges(String databaseName, String tableName, String grantee, Set<PrivilegeGrantInfo> privilegeGrantInfoSet)
     {
         try {
+            Set<HivePrivilegeInfo> existingPrivileges = getTablePrivileges(grantee, databaseName, tableName);
+
+            Set<PrivilegeGrantInfo> privilegesToRevoke = privilegeGrantInfoSet.stream()
+                    .filter(requestedPrivilege -> existingPrivileges.containsAll(parsePrivilege(requestedPrivilege)))
```

Comment:
For example existingPrivileges is 1-elemnt set: ("SELECT")
and requestedPrivileges are ("SELECT","INSERT").
Shouldn't that revoke the "SELECT"?